gnd/src/main/java/com/google/android/gnd/repository/ProjectRepository.java

```diff
@@ -0,0 +1,159 @@
+            .doOnNext(id -> Log.v(TAG, "Requested project id changed: " + id))
+            .switchMap(this::onActivateProjectRequest)
+            .replay(1)
+            .refCount();
+  }
+
+  private Flowable<Loadable<Project>> onActivateProjectRequest(Optional<String> projectId) {
+    // Empty id indicates intent to deactivate the current project. Used on sign out.
+    if (projectId.isEmpty()) {
+      return Flowable.just(Loadable.notLoaded());
+    }
+    String id = projectId.get();
+
+    return syncProjectWithRemote(id)
+        .onErrorResumeNext(error -> getProjectFromLocal(id, error))
+        .doOnSuccess(__ -> localValueStore.setLastActiveProjectId(id))
+        .toFlowable()
+        .compose(Loadable::loadingOnceAndWrap);
+  }
+
+  private Single<Project> syncProjectWithRemote(String id) {
+    return remoteDataStore
+        .loadProject(id)
+        .timeout(LOAD_REMOTE_PROJECT_TIMEOUT_SECS, TimeUnit.SECONDS)
+        .doOnSubscribe(__ -> Log.d(TAG, "Activating project " + id))
+        .doOnSuccess(localDataStore::insertOrUpdateProject);
+  }
+
+  private Single<Project> getProjectFromLocal(String id, Throwable error) {
+    return localDataStore
+        .getProjectById(id)
+        .toSingle()
+        .doOnSubscribe(__ -> Log.d(TAG, "Failed to load project from remote", error));
+  }
+
+  @NonNull
+  public Optional<String> getLastActiveProjectId() {
+    return Optional.ofNullable(localValueStore.getLastActiveProjectId());
+  }
+
+  /**
+   * /** Returns a stream that emits the latest project activation state, and continues to emits
+   * changes to that state until all subscriptions are disposed.
+   */
+  public Flowable<Loadable<Project>> getActiveProjectOnceAndStream() {
+    return activeProjectStream;
+  }
+
+  public void activateProject(String projectId) {
+    Log.v(TAG, "activateProject() called with " + projectId);
+    activateProjectRequests.onNext(Optional.of(projectId));
+  }
+
+  public Flowable<Loadable<List<Project>>> getProjectSummaries(User user) {
+    return loadProjectSummariesFromRemote(user)
+        .onErrorResumeNext(error -> loadProjectSummariesFromLocal(error))
+        .toFlowable()
+        .compose(Loadable::loadingOnceAndWrap);
+  }
+
+  private Single<List<Project>> loadProjectSummariesFromRemote(User user) {
+    return remoteDataStore
+        .loadProjectSummaries(user)
+        .timeout(LOAD_REMOTE_PROJECT_SUMMARIES_TIMEOUT_SECS, TimeUnit.SECONDS)
+        .doOnSubscribe(__ -> Log.d(TAG, "Loading project list from remote"));
+  }
+
+  private Single<List<Project>> loadProjectSummariesFromLocal(Throwable error) {
+    return localDataStore
+        .getProjects()
+        .doOnSubscribe(__ -> Log.d(TAG, "Failed to load project list from remote", error));
+  }
+
+  // TODO: Only return feature fields needed to render features on map.
+  // TODO(#127): Decouple from Project and accept id instead.
+  public Flowable<ImmutableSet<Feature>> getFeaturesOnceAndStream(Project project) {
```

Comment:
Are we exposing this method in two places? Projectrepo and featurerepo? Does this belong in the feature repo only?

Reply:
Wow, good catch. Removing from ProjectRepo in c2fd49688e10ec20ed55a3a92df4fcfdec7279d5.